src/MySqlConnector/MySqlClient/MySqlParameter.cs

```diff
@@ -153,9 +173,16 @@ internal void AppendSqlString(BinaryWriter writer, StatementPreparerOptions opti
 
 		internal static string NormalizeParameterName(string name)
 		{
-			return string.IsNullOrEmpty(name) ? name :
-				name[0] == '?' || name[0] == '@' ? name.Substring(1) :
-				name;
+			name = name.Trim();
+
+			if ((name.StartsWith("@`") || name.StartsWith("?`")) && name.EndsWith("`"))
```

Comment:
Use the overloads that take `StringComparison.Ordinal`. The default behaviour is to be culture-sensitive which is horribly inefficient in CoreCLR on Linux: dotnet/coreclr#5612 dotnet/corefx#5027